src/db/read.rs

```diff
@@ -36,6 +35,28 @@ impl<'a> OwnedRead<'a> {
     }
 }
 
+pub async fn read_commit(
+    whence: Whence,
+    read: &dag::Read<'_>,
+) -> Result<(String, Commit, prolly::Map), ReadCommitError> {
+    use ReadCommitError::*;
+    let hash = match whence {
+        Whence::Hash(s) => s,
+        Whence::Head(s) => read
+            .get_head(&s)
+            .await
+            .map_err(GetHeadError)?
+            .ok_or_else(|| UnknownHead(s.to_string()))?,
```

Comment:
i have to admit the error mapping and ? do make the code a lot more readable than go